idp-binding-service/src/main/java/io/mosip/idp/binding/services/WalletBindingServiceImpl.java

```diff
@@ -0,0 +1,241 @@
+		String publicKey = getJWKString(walletBindingRequest.getPublicKey());
+		publicKeyRegistry.setPsuToken(partnerSpecificUserToken);
+		publicKeyRegistry.setPublicKey(publicKey);
+		publicKeyRegistry.setExpiredtimes(calculateExpiresOn());
+		publicKeyRegistry.setWalletBindingId(walletBindingId);
+		publicKeyRegistry.setCreatedtimes(LocalDateTime.now(ZoneId.of("UTC")));
+		publicKeyRegistry = publicKeyRegistryRepository.saveAndFlush(publicKeyRegistry);
+		
+		IdTokenMapping idTokenMapping = new IdTokenMapping();
+		idTokenMapping.setIdHash(
+				IdentityProviderUtil.generateB64EncodedHash(ALGO_SHA_256, walletBindingRequest.getIndividualId()));
+		idTokenMapping.setPsuToken(partnerSpecificUserToken);
+		idTokenMapping = idTokenMappingRepository.saveAndFlush(idTokenMapping);
+		return publicKeyRegistry;
+	}
+
+	private LocalDateTime calculateExpiresOn() {
+		LocalDateTime currentDateTime = LocalDateTime.now(ZoneId.of("UTC"));
+		return currentDateTime.plusDays(expireDays);
+	}
+
+	private KycAuthResult authenticateIndividual(BindingTransaction bindingTransaction,
+			WalletBindingRequest walletBindingRequest) throws IdPException {
+		List<AuthChallenge> authChallengeList = new ArrayList<AuthChallenge>();
+		authChallengeList.add(walletBindingRequest.getAuthChallenge());
+		KycAuthResult kycAuthResult;
+		try {
+			kycAuthResult = authenticationWrapper.doKycAuth(authPartnerId, apiKey,
+					new KycAuthRequest(bindingTransaction.getAuthTransactionId(),
+							walletBindingRequest.getIndividualId(), authChallengeList));
+		} catch (KycAuthException e) {
+			log.error("KYC auth failed for transaction : {}", bindingTransaction.getAuthTransactionId(), e);
+			throw new IdPException(e.getErrorCode());
+		}
+		if (kycAuthResult == null || (StringUtils.isEmpty(kycAuthResult.getKycToken())
+				|| StringUtils.isEmpty(kycAuthResult.getPartnerSpecificUserToken()))) {
+			log.error("** authenticationWrapper : {} returned empty tokens received **", authenticationWrapper);
+			throw new IdPException(AUTH_FAILED);
+		}
+		return kycAuthResult;
+	}
+
+	private String signWalletBindingId(Map<String, Object> walletBindingIdMap) throws JsonProcessingException {
+
+		String payload = objectMapper.writeValueAsString(walletBindingIdMap);
+		JWTSignatureRequestDto jwtSignatureRequestDto = new JWTSignatureRequestDto();
+		jwtSignatureRequestDto.setApplicationId(Constants.IDP_BINDING_SERVICE_APP_ID);
+		jwtSignatureRequestDto.setReferenceId("");
+		jwtSignatureRequestDto.setIncludePayload(true);
+		jwtSignatureRequestDto.setIncludeCertificate(false);
+		jwtSignatureRequestDto.setDataToSign(IdentityProviderUtil.b64Encode(payload));
+		jwtSignatureRequestDto.setIncludeCertHash(false);
+		JWTSignatureResponseDto responseDto = signatureService.jwtSign(jwtSignatureRequestDto);
+		return responseDto.getJwtSignedData();
+	}
+
+	private String getJWE(Map<String, Object> publicKey, String signWalletBindingId) throws JoseException {
```

Comment:
change the variable name to jwt / signedWalletBindingId

Reply:
fixed